components/_impl/workers/subprocess_rpc.py

```diff
@@ -126,9 +130,17 @@ def _loop(self):
             now = time.time()
             with self._loop_lock:
                 for w_fd, (timeout, start_time) in tuple(self._active_reads.items()):
-                    if now - start_time >= timeout and w_fd in self._active_reads:
-                        os.write(w_fd, _TIMEOUT)
-                        self.pop(w_fd)
+                    if timeout:
+                        if now - start_time >= timeout and w_fd in self._active_reads:
+                            os.write(w_fd, _TIMEOUT)
+                            self.pop(w_fd)
+                    else:
+                        # if timeout is None, only check if the process is still alive
+                        # it could be in zombie status, or has already been reaped
+                        if not psutil.pid_exists(self._proc_id) or \
+                            psutil.Process(self._proc_id).status() == psutil.STATUS_ZOMBIE:
```

Comment:
Random aside: `== psutil.STATUS_ZOMBIE` does such a great job of conveying intent. I love it.